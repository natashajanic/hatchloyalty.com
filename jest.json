{
  "collectCoverage": true,
  "coveragePathIgnorePatterns": ["tests"],
  "globals": {},
  "moduleFileExtensions": [
    "ts",
    "tsx",
    "js"
  ],
  "moduleNameMapper": {
    "^.+\\.(css|less|scss)$": "identity-obj-proxy",
    "\\.(jpg|jpeg|png|gif|eot|otf|webp|svg|ttf|woff|woff2|mp4|webm|wav|mp3|m4a|aac|oga)$": "<rootDir>/tests/assetsTransformer.ts",
    "^src/(.*)$": "<rootDir>/src/$1"
  },
  "setupFiles": [
    "raf/polyfill"
  ],
  "setupFilesAfterEnv": [
    "<rootDir>/tests/setupTests.ts"
  ],
  "snapshotSerializers": [
    "enzyme-to-json/serializer"
  ],
  "transform": {
    "\\.(ts|tsx)$": "ts-jest"
  },
  "testRegex": "/src/.*/__tests__/.*\\.(ts|tsx|js)$",
  "unmockedModulePathPatterns": [],
  "verbose": true
}
