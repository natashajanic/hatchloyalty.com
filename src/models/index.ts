import { FluidObject } from 'gatsby-image'

export interface IBlogPost {
  excerpt: string
  fields: {
    slug: string
  }
  frontmatter: {
    author: string
    date: string
    featuredImage?: IGraphQLImage
    tags: string[]
    title: string
  }
  html: string
}

export interface IGraphQLImage {
  childImageSharp: {
    fluid: FluidObject
  }
}

export interface IRelease {
  changes: Array<{ component: string; description: string; link?: string }>
  date: string
  name: string
}

export interface IResourcePost {
  excerpt: string
  fields: {
    slug: string
  }
  frontmatter: {
    title: string
  }
  html: string
}

export interface ITeamMember {
  headshot?: IGraphQLImage
  name: string
  position: string
}
