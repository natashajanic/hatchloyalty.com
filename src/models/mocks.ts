import { IRelease, IBlogPost, IResourcePost, ITeamMember } from '.'

export const mockBlogPost: IBlogPost = {
  excerpt: 'This is the excerpt for a blog post.',
  fields: {
    slug: '/2019-02-25-test-blog-post/',
  },
  frontmatter: {
    author: 'John Doe',
    date: 'February 25, 2019',
    featuredImage: {
      childImageSharp: {
        fluid: {
          aspectRatio: 1,
          src: 'test/imgs/test-blog-post.png',
          srcSet: '',
          sizes: '',
        },
      },
    },
    tags: ['Test', 'Press Release'],
    title: 'Test Blog Post',
  },
  html: '<div>Test blog post body content.</div>',
}

export const mockRelease: IRelease = {
  changes: [
    { component: 'Test', description: 'Test Change 1' },
    {
      component: 'Test HQ',
      description: 'Test Change 2',
      link: 'https://developer.hatchloyalty.com/test/change-2',
    },
  ],
  date: 'January 2019',
  name: 'Test Release 2019.01',
}

export const mockResourcePost: IResourcePost = {
  excerpt: 'This is the excerpt for a resource post.',
  fields: {
    slug: '/test-resource-post/',
  },
  frontmatter: {
    title: 'Test Resource Post',
  },
  html: '<div>Test resource post body content.</div>',
}

export const mockTeamMember: ITeamMember = {
  headshot: {
    childImageSharp: {
      fluid: {
        aspectRatio: 1,
        src: 'test/imgs/john-doe.png',
        srcSet: '',
        sizes: '',
      },
    },
  },
  name: 'John Doe',
  position: 'CEO',
}
