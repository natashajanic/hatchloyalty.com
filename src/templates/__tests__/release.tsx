import { shallow, ShallowWrapper } from 'enzyme'
import { Link } from 'gatsby'
import 'jest' // tslint:disable-line no-import-side-effect
import * as React from 'react'
import Release from 'src/components/release'
import ReleaseNavigation from 'src/components/release-navigation'
import { mockRelease } from 'src/models/mocks'
import ReleaseTemplate, { IReleaseTemplateProps } from '../release'

describe('ReleaseTemplate', () => {
  let props: IReleaseTemplateProps
  let mountedComponent: ShallowWrapper<any, any, ReleaseTemplate>

  beforeEach(() => {
    props = {
      pageContext: {
        release: mockRelease,
        previous: { ...mockRelease, name: 'Test Previous Release' },
        next: { ...mockRelease, name: 'Test Next Release' },
      },
    }
    mountedComponent = shallow(<ReleaseTemplate {...props} />)
  })

  describe('render', () => {
    it('renders navigation to the next and previous releases', async () => {
      expect(mountedComponent.find(ReleaseNavigation).length).toEqual(1)
      expect(
        mountedComponent.find(ReleaseNavigation).prop('next')
      ).toMatchObject(props.pageContext.next)
      expect(
        mountedComponent.find(ReleaseNavigation).prop('previous')
      ).toMatchObject(props.pageContext.previous)
    })

    it('renders the current release', async () => {
      expect(mountedComponent.find(Release).length).toEqual(1)
      expect(mountedComponent.find(Release).prop('release')).toMatchObject(
        props.pageContext.release
      )
    })

    it('matches the expected snapshot', async () => {
      expect(mountedComponent).toMatchSnapshot()
    })
  })
})
