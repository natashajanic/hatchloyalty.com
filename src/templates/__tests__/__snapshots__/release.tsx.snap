// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ReleaseTemplate render matches the expected snapshot 1`] = `
<Layout
  pageStyle="offWhite"
>
  <SEO
    keywords={
      Array [
        "hatch",
        "hatch loyalty",
        "loyalty",
        "blog",
        "personalization",
        "activation",
        "changelog",
      ]
    }
    title="Changelog"
  />
  <Styled(Component)
    blacklist={
      Array [
        "css",
        "maxWidth",
        "m",
        "mt",
        "mr",
        "mb",
        "ml",
        "mx",
        "my",
        "p",
        "pt",
        "pr",
        "pb",
        "pl",
        "px",
        "py",
      ]
    }
    is="main"
    m="0 auto"
    maxWidth="1024px"
    px={
      Array [
        3,
        4,
      ]
    }
  >
    <ReleaseNavigation
      next={
        Object {
          "changes": Array [
            Object {
              "component": "Test",
              "description": "Test Change 1",
            },
            Object {
              "component": "Test HQ",
              "description": "Test Change 2",
              "link": "https://developer.hatchloyalty.com/test/change-2",
            },
          ],
          "date": "January 2019",
          "name": "Test Next Release",
        }
      }
      previous={
        Object {
          "changes": Array [
            Object {
              "component": "Test",
              "description": "Test Change 1",
            },
            Object {
              "component": "Test HQ",
              "description": "Test Change 2",
              "link": "https://developer.hatchloyalty.com/test/change-2",
            },
          ],
          "date": "January 2019",
          "name": "Test Previous Release",
        }
      }
    />
    <Release
      release={
        Object {
          "changes": Array [
            Object {
              "component": "Test",
              "description": "Test Change 1",
            },
            Object {
              "component": "Test HQ",
              "description": "Test Change 2",
              "link": "https://developer.hatchloyalty.com/test/change-2",
            },
          ],
          "date": "January 2019",
          "name": "Test Release 2019.01",
        }
      }
    />
  </Styled(Component)>
</Layout>
`;
